lib/user_agent/browsers/wondery_crawler.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class UserAgent
+  module Browsers
+    # wondery/develop
+    # wondery/development
+    # wondery/prod
+    # wondery/production
+    # wondery/stage
+    # wondery/staging
+    class WonderyCrawler < Base
+      WONDERY_CRAWLER       = 'Wondery Crawler'
+      WONDERY_CRAWLER_REGEX = /(develop(ment)?)|(stag)|stage|(prod(uction)?)/i.freeze
```

Comment:
You're not using any of these capture groups, so you should make them non-capturing (`?:`) but also you can simplify it greatly:

```
/\A(?:develop|stag|prod)/
```

Reply:
right, what was I thinking?

Reply:
Since we were given a set of `version`s to supprt, would something like this be faster?
`%w[develop staging production].include?(ua.version)`
Of course make the set a constant and freeze

Reply:
@skwongr  that is fine too, however one caveat is it becomes a little susceptible  to changes like from `develop` to `development` or `stage` to `staging`. I am sure the likelihood of it is small, but the overhead of simple regex testing is small also, i thought it would not hurt.

Reply:
> the overhead of simple regex testing is small

We can very easily test that:

```
irb(main):001:0> require 'benchmark'
=> true
irb(main):002:0> require 'set'
=> false
irb(main):003:0> WONDERY_CRAWLER_VERSIONS = Set.new(%w[develop development stage staging prod production]).freeze
=> #<Set: {"develop", "development", "stage", "staging", "prod", "production"}>
irb(main):004:0> WONDERY_CRAWLER_REGEX = /\A(?:develop|stag|prod)/.freeze
=> /\A(?:develop|stag|prod)/
irb(main):005:0> matching_string = 'stage'
=> "wondery/stage"
irb(main):006:0> not_matching_string = 'ios'
=> "wondery/ios"
irb(main):007:0> iterations = 100_000_000
=> 100000000
irb(main):008:1* Benchmark.bmbm(10) do |bm|
irb(main):009:1*   bm.report('set:') { iterations.times { WONDERY_CRAWLER_VERSIONS.include?(matching_string); WONDERY_CRAWLER_VERSIONS.include?(not_matching_string) } }
irb(main):010:1*   bm.report('regex:') { iterations.times { matching_string.match?(WONDERY_CRAWLER_REGEX); not_matching_string.match?(WONDERY_CRAWLER_REGEX) } }
irb(main):011:0> end
Rehearsal ----------------------------------------------
set:        15.707671   0.080513  15.788184 ( 15.812922)
regex:      18.000804   0.090924  18.091728 ( 18.117390)
------------------------------------ total: 33.879912sec

                 user     system      total        real
set:        15.539771   0.079985  15.619756 ( 15.641926)
regex:      18.848026   0.104020  18.952046 ( 18.978743)
=> [#<Benchmark::Tms:0x00007f88fc00faf0 @label="set:", @real=15.641926000011154, @cstime=0.0, @cutime=0.0, @stime=0.07998500000000008, @utime=15.539771000000002, @total=15.619756000000002>, #<Benchmark::Tms:0x00007f88fc00f910 @label="regex:", @real=18.97874299995601, @cstime=0.0, @cutime=0.0, @stime=0.10402, @utime=18.84802599999999, @total=18.95204599999999>]
irb(main):012:0> '%04.4f%%' % ((18.978743-15.641926)/18.978743  * 100)
=> "17.5819%"
```

So we can round up and say that `Set#include?` is 18% faster. This would be run for every user agent where we didn't find a match yet, and if Wondery adds another suffix to their crawler, adding it to that set is trivial. So I would say that the Set approach is better.